pandas/tests/test_resample.py

```diff
@@ -722,35 +722,19 @@ def series_and_frame(self, request, index):
         if request.param == DataFrame:
             return self.frame(index)
 
-    def test_asfreq_downsample(self):
-        s = self.create_series()
-
-        result = s.resample('2D').asfreq()
-        expected = s.reindex(s.index.take(np.arange(0, len(s.index), 2)))
-        expected.index.freq = to_offset('2D')
-        assert_series_equal(result, expected)
-
-        frame = s.to_frame('value')
-        result = frame.resample('2D').asfreq()
-        expected = frame.reindex(
-            frame.index.take(np.arange(0, len(frame.index), 2)))
-        expected.index.freq = to_offset('2D')
-        assert_frame_equal(result, expected)
-
-    def test_asfreq_upsample(self):
-        s = self.create_series()
-
-        result = s.resample('1H').asfreq()
-        new_index = self.create_index(s.index[0], s.index[-1], freq='1H')
-        expected = s.reindex(new_index)
-        assert_series_equal(result, expected)
+    @pytest.mark.parametrize('freq', ['2D', '1H'])
+    def test_asfreq(self, series_and_frame, freq):
```

Comment:
@jreback 
I consolidated the ```test_asfreq_downsample()``` and ```test_asfreq_upsample()``` tests in the ```Base``` class as well (so we don't have to override them in ```TestPeriodIndex```). PTAL if everything's still correct and covered as before.